### DISCUSSION TITLE
Sony XA2 (pioneer)

### DISCUSSION THREAD

**Discussion by someone30** [Divested-Mobile/DivestOS-Build#162] [General] [answered]
I can confirm: the installation of DivestOS October version works on the Sony XA2 (pioneer). The performance is rather sluggish - as expected with this Android version / Hardware combination.

Two hiccups during installation though:

1. `fastboot update divested-19.1-20221008-dos-pioneer-fastboot.zip` fails:

> Device product is ‘H3113’.
> Update requires ‘sdm660’

2. Signature verification fails when flashing via LineageOS recovery (expected behavior?)

What do you think: how big are the chances relocking the bootloader will work with this device? I would love to try. 🙏

**Comment by SkewedZeppelin** [comment, marked as answer]
>  Update requires ‘sdm660’

Some devices do this. You can either extract the zip and flash each partition manually or just use a recovery and update via sideload.

> Signature verification fails when flashing via LineageOS recovery (expected behavior?)

That is expected, as their recovery checks for their keys.

If you use the DivestOS recovery you shouldn't have such issue.

> relocking the bootloader will work with this device

I don't think the chances are high, but it may be possible. Just need someone willing to destroy their device to test it...

**Reply by someone30** [reply]
> > Update requires ‘sdm660’
> 
> Some devices do this. You can either extract the zip and flash each partition manually or just use a recovery and update via sideload.

To avoid the mentioned error messages for new XA2/pioneer users could we just edit the android-info.txt file of the DivestOS fastboot zip and replace "sdm660" with "H3113"? Or must the ID also be known by the fastboot binary?

If that's not possible could you provide a recovery button/link instead for this device on the "Device Downloads" page? The status could be updated to "Reported Working" on this occasion.

> > Signature verification fails when flashing via LineageOS recovery (expected behavior?)
> 
> That is expected, as their recovery checks for their keys.
> 
> If you use the DivestOS recovery you shouldn't have such issue.

Confirmed. Used the boot.img recovery of the DivestOS fastboot zip - worked flawlessly.

**Reply by SkewedZeppelin** [reply]
> To avoid the mentioned error messages for new XA2/pioneer users could we just edit the android-info.txt file of the DivestOS fastboot zip and replace "sdm660" with "H3113"?

You could do this, but most users don't/shouldn't have to repack the 4GB zip file.

> could you provide a recovery

You can extract the recovery from the fastboot.zip.

I could make recovery always available, but that isn't enough for some devices (boot+dtbo partition): https://github.com/Divested-Mobile/DivestOS-Build/blob/master/Scripts/Common/Functions.sh#L365
